modules/godot/databags/databag_timer.cpp

```diff
@@ -0,0 +1,66 @@
+#include "databag_timer.h"
+
+void TimersDatabag::_bind_methods() {
+}
+
+void TimersDatabag::set_now(const uint64_t p_now) {
+	now = p_now;
+}
+
+bool TimersDatabag::is_valid_timer_handle(const godex::TimerHandle p_handle) const {
+	return timers.size() < p_handle.timerIndex && timers[p_handle.timerIndex].generation == p_handle.generation && timers[p_handle.timerIndex].endTime > 0;
+}
+
+godex::TimerHandle TimersDatabag::new_timer(const real_t p_end_in_seconds) {
+	return new_precise_timer(static_cast<uint64_t>(p_end_in_seconds * 1'000'000.0));
+}
+
+godex::TimerHandle TimersDatabag::new_precise_timer(const uint64_t p_end_in_microseconds) {
+	ERR_FAIL_COND_V_MSG(p_end_in_microseconds == 0, godex::TimerHandle(), "Can't make a new timer that ends at 0, it's invalid.");
+	if (destroyed_timers.size() > 0) {
+		godex::TimerHandle finalHandle = destroyed_timers[destroyed_timers.size() - 1];
+		internal_restart_precise_timer(finalHandle, now + p_end_in_microseconds);
+		destroyed_timers.remove_unordered(destroyed_timers.size() - 1);
+		return finalHandle;
+	} else {
+		timers.push_back(godex::Timer(now + p_end_in_microseconds, 0));
+		return godex::TimerHandle(timers.size() - 1, 0);
+	}
+}
+
+void TimersDatabag::restart_timer(const godex::TimerHandle p_timer_handle, const real_t p_end_in_seconds) {
+	restart_precise_timer(p_timer_handle, static_cast<uint64_t>(p_end_in_seconds * 1'000'000.0));
+}
+
+void TimersDatabag::restart_precise_timer(const godex::TimerHandle p_timer_handle, const uint64_t p_end_in_microseconds) {
+	ERR_FAIL_COND_MSG(!is_valid_timer_handle(p_timer_handle), "The handle" + itos(p_timer_handle.timerIndex) + " is not a valid handle.");
+	internal_restart_precise_timer(p_timer_handle, p_end_in_microseconds);
+}
+
+bool TimersDatabag::is_done(const godex::TimerHandle p_timer_handle) const {
+	ERR_FAIL_COND_V_MSG(!is_valid_timer_handle(p_timer_handle), true, "The handle" + itos(p_timer_handle.timerIndex) + " is not a valid handle.");
+	return timers[p_timer_handle.timerIndex].endTime <= now;
+}
+
+void TimersDatabag::abort_timer(const godex::TimerHandle p_timer_handle) {
```

Comment:
I prefer call this `destroy_timer`. `abort_timer` seems like "pause" the current timer but keep it valid.